tests/v2/validation/standalone/corral_generic_test.go

```diff
@@ -0,0 +1,69 @@
+package standalone
+
+import (
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/corral"
+
+	"github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	defaultRandStringLength = 5
+)
+
+type CorralStandaloneTestSuite struct {
+	suite.Suite
+	session *session.Session
+}
+
+func AppendRandomString(baseClusterName string) string {
+	clusterName := "auto-" + baseClusterName + "-" + namegenerator.RandStringLower(defaultRandStringLength)
+	return clusterName
+}
+
+func (r *CorralStandaloneTestSuite) TearDownSuite() {
+	r.session.Cleanup()
+}
+
+func (r *CorralStandaloneTestSuite) SetupSuite() {
+	testSession := session.NewSession(r.T())
+	r.session = testSession
+
+	corralConfig := corral.CorralConfigurations()
+	err := corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser)
+	require.NoError(r.T(), err, "error reading corral configs")
+}
+
+func (r *CorralStandaloneTestSuite) TestGenericCorralPackage() {
+	corralPackage := corral.CorralPackagesConfig()
+	newPackages := []string{}
+	for key, value := range corralPackage.CorralPackagePath {
+		r.T().Log("key: ", key, "value: ", value)
+		newPackageName := AppendRandomString(key)
+		newPackages = append(newPackages, newPackageName)
+		r.T().Log("Running Corral For: ", newPackageName)
+		corralRun, err := corral.CreateCorral(newPackageName, value, corralPackage.Debug, corralPackage.Cleanup)
+		// if cleanup is enabled, delete packages that fail to correctly provision
+		if err != nil && corralPackage.Cleanup {
+			r.T().Log("Error Running Corral, attempting to clean up resources...  ")
+			corral.DeleteCorral(newPackageName)
+		}
+		require.NoError(r.T(), err, "error creating corral %v", key)
+		require.NotNil(r.T(), corralRun, "corral run had no output")
+	}
+	// if cleanup is enabled, delete packages in reverse order that they were created in
+	if corralPackage.Cleanup {
```

Comment:
Why don't we use this Cleanup bool in conjunction with our session cleanup ie. when we create a corral we register it's delete function and when we defer session.Cleanup() it will also clean up the corral. Do you agree?